lola.html

```diff
@@ -0,0 +1,953 @@
+
+      <p>The server-to-server communications involved in this workflow do need to be reliable (URLs, messages and schemas) in order to be interoperable, so those sections are normative.</p>
+
+    </section>
+
+
+    <section id="sotd">
+      <p>This document is an individual submission to the SocialWeb Incubator Community Group, which has formed a task force to focus on the work of data portability.  Interested parties should join the public-swicg@w3.org mailing list to discuss.</p>
+
+      <p>This document is compatible with the ActivityPub Data Portability document, which describes a way to perform an offline account migration.</p>
+
+    </section>
+
+    <section class="informative">
+      <h2>Approach</h2>
+      <p>The technical approach proposed for LOLA is, at a high level: </p>
+      <ul>
+        <li>Source server advertises an OAuth endpoint for authorizing account portability (this can co-exist with other OAuth endpoints e.g.  for authenticating the user).</li>
+        <li>Destination server initiates OAuth to gain user authorization and give the destination server a secure token for source server access requests</li>
+        <li>Destination server can use the secure token and find the right endpoints to start copying data from the source</li>
+        <li>The focus is on copying content (rather than a changelog).  The destination may populate an ActivityPub outbox.</li>
+        <li>Content object IDs may be generated in any mechanism the destination server chooses</li>
+        <li>Destination may also copy additional activities, media and allow/block/follow lists</li>
+        <li>Private items are in scope if the user authorizes</li>
+        <li>Optionally, if the user asks later, the source server can notify follows of the source account that the account has moved and what 3rd party recipients can do with these notifications</li>
+        <li>Optionally, if the user asks later, the source server can redirect requests for content with URLs for the destination server.</li>
+      </ul>
+
+    </section>
+
+    <section class="informative">
+      <h2>Use cases</h2>
+
+
+      <p>The primary user need to solve with LOLA is to allow Fediverse users to move their accounts in response to moderation and defederation decisions.  The connections between servers and the communities on servers form a structure that has close connections, looser connections and blocks, allowing many different types of communities to co-exist.  That does put a lot of pressure on users to be able to move their accounts (due to moderation decisions they disagree with, because of server defederation, or just to move closer to an affinity group).   A user making such an account move will sometimes want to copy their previously posted content to their new location, sometimes follow the same people, and sometimes notify their followers so their followers can resubscribe in a new location.  Servers implementing LOLA can provide these choices to users.</p>
```

Comment:
```suggestion
      <p>The primary user need to solve with LOLA is to allow Fediverse users to move their accounts in response to moderation and defederation decisions.  The connections between servers and the communities on servers form a structure that has close connections, looser connections, and explicit blocks, allowing many different types of communities to co-exist.  That does put a lot of pressure on users to be able to move their accounts (due to moderation decisions they disagree with, because of server defederation, or just to move closer to an affinity group).   A user making such an account move will sometimes want to copy their previously posted content to their new location, sometimes follow the same people, and sometimes notify their followers so their followers can resubscribe in a new location.  Servers implementing LOLA can provide these choices to users.</p>
```